test/cartesian/ZAxis.spec.tsx

```diff
@@ -28,6 +28,146 @@ describe('<ZAxis />', () => {
       yAxis: 1400,
     },
   ];
+
+  it('renders ZAxis with correct domain and range', () => {
+    const { container } = render(
+      <ScatterChart width={400} height={400}>
+        <XAxis dataKey="xAxis" />
+        <YAxis dataKey="yAxis" />
+        <ZAxis dataKey="zAxis" range={[100, 1000]} />
+        <Scatter data={data} />
+      </ScatterChart>,
+    );
+
+    // Instead of accessing __data__, directly check SVG elements or use testing-library utilities
+    const zAxisTicks = container.querySelectorAll('.recharts-cartesian-axis-tick');
```

Comment:
I don't think these are necessarily z axis ticks - they're probably not since z axis doesn't have any ticks. Since z axis changes scatter size we'll have to assert that somehow